code/modules/mob/living/silicon/decoy/decoy.dm

```diff
@@ -31,16 +34,17 @@
 
 	update_stat()
 
-/mob/living/silicon/decoy/death(gibbed, deathmessage = "sparks up and falls silent...")
+/mob/living/silicon/decoy/death(gibbed, deathmessage)
 	set waitfor = 0
-	if(stat == DEAD) return FALSE
+	. = ..()
```

Comment:
Do you know what this does? Do you want the AI to inherit the death behavior of other silicons?

Reply:
I don't but ninja said I shouldn't be overwriting the proc like that.

Reply:
VSC allows you to right-click and `Go to Definition` (var redefinition in wrong places borks this). You can do this all the way down to the first definition.
What I'm asking you here is that you understand the changes you are making. Else I have to do the kind of thing I described every time (I'm likely to do it anyway, though).
In here you are changing calling the parent proc before your code is processed, instead of after.

Reply:
I need to call it before since the code for the death sets it's canmove to a state where anyone can move through the dead sprite, which I later set so noone can.